### PR TITLE
AP-9987 # add element properties regarding visiblity and editability …

### PR DESCRIPTION
**Pull request by divporter**
…for approvers

## Requester Checklist

Please only check the items that you have actioned. Do not check items that are not applicable to your PR.

### Implementation

- [ ] Have you tested your implementation locally?
- [ ] If applicable, has an appropriate changelog entry been added? Do not include if you are fixing/changing something that is only in the current release.
- [ ] There are no warnings that have been suppressed unnecessarily
- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Have all OneBlink dependency updates been completed (eg apps / apps-react / types etc).
- [ ] Have you ensured this change does not add unwanted dependencies?
- [ ] If this PR contains a refactor, have relevant Jira testing tasks added?
- [ ] Changes that will knowingly make the feature/bug incomplete have been commented with TODO and a description of what needs to be done to finish the feature/bug
- [ ] Any changes made to public APIs have been reflected in the documentation
- [ ] Have you isolated business logic where possible to allow for unit testing?

### Logging and Debugging

- [ ] Are the error messages, if any, informative?
- [ ] Are there enough log events and are they written in a way that allows for easy debugging?
- [ ] "Debugging" code removed
- [ ] Front-end: No erroneous `Console.WriteLines`

### Readability

- [ ] All class, variable, property and method modifiers are provided with the smallest scope possible
- [ ] New files, variables and functions are descriptive/comprehensible and named consistently.
- [ ] There is no dead code (unreachable code)
- [ ] There is no usage of [magic numbers](<https://en.wikipedia.org/wiki/Magic_number_(programming)>)
- [ ] There is no commented out code.
- [ ] In hard-to-understand areas, comments exist and describe rationale or reasons for decisions in code

### Security

- [ ] All personal data inputs are checked (for the correct type, length/size, format, and range).
- [ ] No sensitive information is logged or visible in a stacktrace
- [ ] Are authorization and authentication handled correctly?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Do API endpoints return appropriate status codes

## Reviewer Guide

- It is important that you understand the purpose of the PR.
- You are encouraged to engage with the requestor if you do not understand any of the proposed code changes/additions/deletions.
- Do you, the reviewer, understand what the code does? Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be accepted?
- Is a framework, API, library, or service used that should not be used? Are there alternatives you could recommend?
- Are there existing hooks/components/functions in the same code base that could be utilised?
- When reviewing tests, attempt to identify missing edge cases that may be relevant to the proposed implementation
- Ensure you check for:
  - Security
  - Scalability
  - Performance
  - Maintainability

---

> [!NOTE]
> **Low Risk**
> Changes are limited to TypeScript declarations; consumers must implement runtime behavior separately, with low direct production risk from this PR alone.
> 
> **Overview**
> Extends form element definitions so builders and runtimes can control **approver** behavior separately from end-user `isHidden` and `readOnly`.
> 
> On every element’s base shape, adds optional **`isVisibleToApprovers`** to control whether approvers see the field. Introduces **`FormElementEditableByApprovers`** with **`editableByApprovers`** and mixes it into input-style elements (text, dates, choices, files, addresses, draw/camera, repeatable sets, and similar) so approver editability can be configured per field.
> 
> Also normalizes many JSDoc comments to single-line form in `forms.d.ts` (no behavioral type changes).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8d07ede4c99964dcfd3dccda5b0fb6c9ee2afd24. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: typescript/forms.d.ts [hunk 1]
```diff
@@ -123,28 +124,26 @@ export type LookupFormElement = FormElementBase & {
...
-   * The validation message to display if this input requires a value entered by
-   * the user.
-   */
+  /** The validation message to display if this input requires a value entered by the user. */
   requiredMessage?: string
 }
 
+export type FormElementEditableByApprovers = {
+  /** Determine if this input can be edited by approvers (`true`) or not (`false`). */
+  editableByApprovers?: boolean
```
Comment (mymattcarroll) [inline typescript/forms.d.ts:144]: Closing as this does not support adding a whitelist of "approval steps"  in the future. New PR incoming.
